content/6_Advanced/Wavelet.mdx

```diff
@@ -32,9 +30,232 @@ Like a segment tree on values rather than indices.
 	</Resource>
 </Resources>
 
-### Solution - Range K-th Smallest
+Suppose you want to support the following queries:
```

Comment:
why was this solution header deleted?

if you want to explain how a wavelet tree works before an explanatory problem, move it up

Reply:
The example problem is essentially a second type of query, but to explain it more clearly, I need first to describe the first type of query

Reply:
you should make that clear from the start 👀 

also there should still be a solution header for the problem. you would probably move the first type of query and its explanation before this header.

Reply:
If I understood correctly, the structure should be as follows:

- First, we will discuss a different type of query to clarify the second.
= Explanation of the first type of query.
- Solution header followed by the explanation of the second type of query.

Reply:
you don't have the solution header at the moment—if you think explaining the first query is necessary to understand the second query, i would suggest a structure like so:


* solution header
* before we solve this problem, let's consider a simpler ver. where ...
* explanation of first query
* bring it back to second query
* explain
* conclude the problem
* implementation


also if you could integrate some of the info blocks in the module that'd be great, right now they kind of interrupt the flow of the content. they generally should only appear once at the beginning or end of a section.

Reply:
The info blocks were once part of the text, but since it is not necessary to read them to understand, I have moved them to info blocks. However, I believe they could provide them for those who need to recap or didn't full understand what was said.
Also the structure was changed.